metrics.go

```diff
@@ -22,20 +32,16 @@ func Timer(key string) *timer {
 }
 
 func (t *timer) Emit() {
-	if !enabled {
+	if !publishEnabled {
 		return
 	}
-
-	go func(key string, start, end time.Time) {
-		select {
-		case statQueue <- NewTimer(
-			fmt.Sprintf("%s.%s", metricPrefix, key),
-			start,
-			end,
-		):
-		default:
-		}
-	}(t.key, t.start, time.Now())
+	timer := NewTimer(
+		fmt.Sprintf("%s.%s", metricPrefix, t.key),
+		t.start,
+		time.Now(),
+	)
+	publishWG.Add(1)
```

Comment:
Would there be a problem moving the Add(1) into the publish method?  A little cleaner